app/services/activities/import_from_csv.rb

```diff
@@ -181,20 +185,22 @@ def initialize(activity, row)
         @row = row
       end
 
-      def build
-        ImplementingOrganisation.find_or_initialize_by(
-          activity_id: activity.id
-        ) { |implementing_organisation|
-          implementing_organisation.name = row["Implementing organisation name"]
-          implementing_organisation.reference = row["Implementing organisation reference"]
-          implementing_organisation.organisation_type = row["Implementing organisation sector"]
-        }
+      def organisation
+        @organisation ||= begin
+          ImplementingOrganisation.find_or_initialize_by(
+            activity_id: activity.id
+          ) { |implementing_organisation|
+            implementing_organisation.name = row["Implementing organisation name"]
+            implementing_organisation.reference = row["Implementing organisation reference"]
+            implementing_organisation.organisation_type = row["Implementing organisation sector"]
+          }
+        end
       end
 
       def add_errors(errors)
         errors.delete(:implementing_organisations)
 
-        implementing_organisation_errors = @activity.implementing_organisations.first.errors.messages
+        implementing_organisation_errors = organisation.errors.messages
```

Comment:
Won't `organisation.errors` always be empty (`organisation.valid?` isn't called)?

I think there's a test case missing for when the activity's implementing_organisation is invalid 🤔

Reply:
Ah, good call. I'll tweak this

Reply:
@leeky Think this is ready for another look